vnext/Microsoft.ReactNative/Views/ViewPanel.cpp

```diff
@@ -127,12 +127,25 @@ winrt::AutomationPeer ViewPanel::OnCreateAutomationPeer() {
 
 /*static*/ void ViewPanel::SetTop(xaml::UIElement const &element, double value) {
   element.SetValue(TopProperty(), winrt::box_value<double>(value));
-  element.InvalidateArrange();
+  InvalidateForArrange(element);
 }
 
 /*static*/ void ViewPanel::SetLeft(xaml::UIElement const &element, double value) {
   element.SetValue(LeftProperty(), winrt::box_value<double>(value));
-  element.InvalidateArrange();
+  InvalidateForArrange(element);
+}
+
+void ViewPanel::InvalidateForArrange(xaml::UIElement element) {
+  // If the element's position has changed, we must invalidate the parent for arrange,
+  // as it's the parent's responsibility to arrange its children.
+  auto parent = VisualTreeHelper::GetParent(element);
+  if (parent) {
+    parent.try_as<xaml::UIElement>().InvalidateArrange();
```

Comment:
>parent.try_as<xaml::UIElement>().InvalidateArrange(); [](start = 4, length = 53)

Note that we have other scenarios that might be broken, due to us having more than one element in the hierarchy for a View.  The parent of a ViewPanel might not itself be a ViewPanel.  I find this code confusing!  I'm choosing not to address that potential issue here, to make it safer for us to backport this fix, and do more research first.